### PR TITLE
adjust matlab inplace modification in main comparison table

### LINKED ISSUE #70: The "in-place modification" for MATLAB is wrong
IN the cheatsheet:

```
function f(out, x)
     out = x.^2
end
x = rand(10)
y = zeros(length(x), 1)
f(y, x)
```

Multiple problems with it:

- This is not valid MATLAB syntax. Running this code snippet throws an error as there are statemnts outside a function
- Even if you did define such a function, MATLAB has no concept of inplace modification. Such a thing isn't possible (without magic like `assignin`)

### PR DESCRIPTION
**Pull request by mmcky**
fixes #70

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -442,12 +442,12 @@ Programming
     |                         |     f = @(x) a + x                   |        return a + x                   |        f(x) = a + x           |
     |                         |     f(1.0)                           |    f(1.0)                             |        f(1.0)                 |
     +-------------------------+--------------------------------------+---------------------------------------+-------------------------------+
-    |                         | .. code-block:: matlab               | .. code-block:: python                | .. code-block:: julia         |
+    |                         |  No simple syntax to achieve this    | .. code-block:: python                | .. code-block:: julia         |
```
Comment (jlperla) [inline index.rst]: Sounds good to me.

the only alternative to consider is  `No consistent or simple syntax to achieve this<https://blogs.mathworks.com/loren/2007/03/22/in-place-operations-on-data/>`___

In some ways, looking at this confusion is a good sales job for python or julia!

Comment (mmcky) [reply index.rst]: thanks @jlperla :-)

I'll try and add the link. Table formatting in RST (well any simple markup really) is awful :-)

Comment (jlperla) [reply index.rst]: Can't wait for myst :-)  If it takes you more than about 3 minutes it probably isn't worth it.  We should stick with your change.

Comment (mmcky) [reply index.rst]: Done

**Pull request merged**